champions/src/main/java/me/mykindos/betterpvp/champions/listeners/ArrowListener.java

```diff
@@ -44,51 +47,52 @@ public ArrowListener(Champions champions) {
         this.champions = champions;
     }
 
-    @EventHandler
-    public void onShootBow(EntityShootBowEvent event) {
-        if (event.getProjectile() instanceof Arrow arrow) {
-            if (event.getEntity() instanceof Player player) {
-                arrow.setMetadata("ShotWith", new FixedMetadataValue(champions, player.getInventory().getItemInMainHand().getType().name()));
-                arrows.put(arrow, event.getForce() / 3);
-            }
-        }
+    private boolean isArrow(Projectile projectile) {
+        return projectile instanceof Arrow || projectile instanceof SpectralArrow;
     }
 
     @EventHandler(priority = EventPriority.LOWEST)
-    public void onBaseArrowDamage(DamageEvent event) {
-        if (event.getProjectile() instanceof Arrow || event.getProjectile() instanceof SpectralArrow) {
-            event.setDamage(baseArrowDamage);
+    public void onShootBow(EntityShootBowEvent event) {
+        if (!isArrow((Projectile) event.getProjectile())) {
+            return;
         }
+
+        final AbstractArrow arrow = (AbstractArrow) event.getProjectile();
+        if (!(event.getEntity() instanceof Player player)) {
+            return;
+        }
+
+        final String item = player.getInventory().getItemInMainHand().getType().name();
+        arrow.setMetadata("ShotWith", new FixedMetadataValue(champions, item));
+        arrow.setMetadata("Force", new FixedMetadataValue(champions, event.getForce()));
+        if (critArrowDamage <= 0) {
+            arrow.setCritical(false);
+        }
+        arrow.setDamage(baseArrowDamage);
     }
 
     @EventHandler(priority = EventPriority.LOWEST)
     public void onProjectileDelay(PreDamageEvent event) {
-        if (event.getDamageEvent().getProjectile() == null) {
+        final DamageEvent damageEvent = event.getDamageEvent();
+        final Projectile projectile = damageEvent.getProjectile();
+        if (projectile == null) {
             return;
         }
 
-        event.getDamageEvent().setForceDamageDelay(0);
-        event.getDamageEvent().setDamageDelay(0);
-    }
-
-    @EventHandler(priority = EventPriority.HIGHEST)
-    public void onArrowDamage(DamageEvent event) {
-        if (event.getProjectile() instanceof Arrow arrow) {
-            if (arrows.containsKey(arrow)) {
-                float dmgMultiplier = arrows.get(arrow);
-                event.setDamage(event.getDamage() * dmgMultiplier);
-                arrows.remove(arrow);
-                if (arrow.isValid()) {
-                    arrow.remove();
-                }
-            }
+        if (isArrow(projectile)) {
+            final AbstractArrow arrow = (AbstractArrow) projectile;
+            final float force = (float) Objects.requireNonNull(projectile.getMetadata("Force").getFirst().value());
+            arrow.remove();
+            damageEvent.setDamage((arrow.getDamage() * force / 3) + (arrow.isCritical() ? critArrowDamage : 0.0));
```

Comment:
This needs to be done when the damage event is processed, otherwise risk conflicting with other events with a HIGHEST priority